WebDriverAgentLib/Categories/XCElementSnapshot+FBHelpers.m

```diff
@@ -97,11 +99,14 @@ - (XCElementSnapshot *)fb_parentMatchingType:(XCUIElementType)type
   return snapshot;
 }
 
-- (XCElementSnapshot *)fb_parentMatchingOneOfTypes:(NSArray<NSNumber *> *)types
+- (XCElementSnapshot *)fb_findVisibleParentMatchingOneOfTypes:(NSArray<NSNumber *> *)types
 {
   XCElementSnapshot *snapshot = self.parent;
-  while (snapshot && !isSnapshotTypeAmongstGivenTypes(snapshot, types)) {
-      snapshot = snapshot.parent;
+  while (snapshot) {
+    if (isSnapshotTypeAmongstGivenTypes(snapshot, types) && [snapshot isWDVisible] && doesSnapshotHasMoreThanOneVisibleChildrenSnapshots(snapshot)) {
```

Comment:
I added `doesSnapshotHasMoreThanOneVisibleChildrenSnapshots ` as i ran into issue where it failed because parent element didn't have more than 1 children snapshot. Let me know if you have alternate suggestion for this. I can change this name `fb_findVisibleParentMatchingOneOfTypes` to `fb_findVisibleParentMatchingOneOfTypesAndHasMoreThanOneVisiableChildren`. Also is it worth to send `visibleCellSnapshots` and `cellSnapshots` along with `scrollview` , together as struct so that we don't have to re-calculate.